packages/driver/src/cy/commands/navigation.ts

```diff
@@ -969,24 +982,36 @@ export default (Commands, Cypress, cy, state, config) => {
           return cannotVisitDifferentOrigin(remote.origin, previousDomainVisited, remote, existing, options._log)
         }
 
-        const current = $Location.create(win.location.href)
-
-        // if all that is changing is the hash then we know
-        // the browser won't actually make a new http request
-        // for this, and so we need to resolve onLoad immediately
-        // and bypass the actual visit resolution stuff
-        if (bothUrlsMatchAndOneHasHash(current, remote)) {
-          // https://github.com/cypress-io/cypress/issues/1311
-          if (current.hash === remote.hash) {
-            consoleProps['Note'] = 'Because this visit was to the same hash, the page did not reload and the onBeforeLoad and onLoad callbacks did not fire.'
+        // in multi-domain, the window may not have been set yet if nothing has been loaded in the secondary domain,
+        // it's also possible for a new test to start and for a cross-domain failure to occur if the win is set but
+        // the AUT hasn't yet navigated to the secondary domain
+        if (win) {
+          try {
+            const current = $Location.create(win.location.href)
+
+            // if all that is changing is the hash then we know
+            // the browser won't actually make a new http request
+            // for this, and so we need to resolve onLoad immediately
+            // and bypass the actual visit resolution stuff
+            if (bothUrlsMatchAndOneHasHash(current, remote)) {
+              // https://github.com/cypress-io/cypress/issues/1311
+              if (current.hash === remote.hash) {
+                consoleProps['Note'] = 'Because this visit was to the same hash, the page did not reload and the onBeforeLoad and onLoad callbacks did not fire.'
+
+                return onLoad({ runOnLoadCallback: false })
+              }
 
-            return onLoad({ runOnLoadCallback: false })
+              return changeIframeSrc(remote.href, 'hashchange')
+              .then(() => {
+                return onLoad({})
+              })
+            }
+          } catch (e) {
+            // if this is a cross-domain error, skip it
+            if (e.name !== 'SecurityError') {
```

Comment:
Is this to handle the case [above](https://github.com/cypress-io/cypress/pull/20467/files#diff-5e3068df8d362da490659f784bf53031dc3be4abc2fb4b09e7989b3d44b38ca0R986)?

Reply:
This is to handle the scenario where `Cypress.state('window')` in the spec bridge is on the secondary but a new test has started and the AUT hasn't navigated to the secondary yet so `win.location.href` throws a cross-origin error.